### PR TITLE
Update TiDB architecture

### PR DESCRIPTION
**Pull request by uglyengineer**
### What is changed, added or deleted? (Required)

Update content for `architecture.md`

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3222 
- Other reference link(s):

### CODE REVIEW

**Comment by uglyengineer** [comment]
@SunRunAway @TomShawn  PTAL, thanks!

File: architecture.md [hunk 1]
```diff
@@ -7,54 +7,28 @@ aliases: ['/docs/dev/architecture/']
...
-The TiDB platform is comprised of three key components: the TiDB server, the PD server, and the TiKV server. In addition, TiDB also provides the [TiSpark](https://github.com/pingcap/tispark/) component for complex OLAP requirements and the [TiDB Operator](https://docs.pingcap.com/tidb-in-kubernetes/v1.1/tidb-operator-overview) to make things simpler for the deployment and management on cloud.
+Compared to traditional databases, TiDB has the following advantages: 
+* Has distributed architecture and supports elastic scalability, flexible capacity expansion & reduction
+* Fully compatible with the MySQL protocol and the common features and syntax of MySQL, able to directly replicate from MySQL in most scenarios
+* Supports high availability in default and executes data recovery and failover autometicly; transparent to the business
+* Supports ACID transaction, suitable for scenarios requiring strong consistency such as bank transfer
+* A rich toolchain ecosystem covers a variety of scenarios including data migration, synchronization, backup, etc.
 
-![image alt text](/media/tidb-architecture.png)
+![image alt text](/media/tidb-architecture-1.png)
```
Comment (lilin90) [inline architecture.md:17]: This is a broken link.
```suggestion
![TiDB Architecture](/media/tidb-architecture-1.png)
```

File: architecture.md [hunk 2]
```diff
@@ -7,54 +7,28 @@ aliases: ['/docs/dev/architecture/']
 
 # TiDB Architecture
 
-The TiDB platform is comprised of three key components: the TiDB server, the PD server, and the TiKV server. In addition, TiDB also provides the [TiSpark](https://github.com/pingcap/tispark/) component for complex OLAP requirements and the [TiDB Operator](https://docs.pingcap.com/tidb-in-kubernetes/v1.1/tidb-operator-overview) to make things simpler for the deployment and management on cloud.
+Compared to traditional databases, TiDB has the following advantages:
```
Comment (lilin90) [inline architecture.md:10]:
```suggestion
Compared to traditional databases, TiDB has the following advantages:

```

File: architecture.md [hunk 3]
```diff
@@ -7,54 +7,28 @@ aliases: ['/docs/dev/architecture/']
...
 
-1. Receiving the SQL requests
+## Placement Driver (PD) server
 
-2. Processing the SQL related logics
+The PD server is the managing component of the entire cluster and is in charge of storing real-time data distribution of every single TiKV node and overall topology structure of the entire TiDB cluster, providing Dashboard management UI, and allocating transaction IDs. The PD server can be seen as ‘The Brain’ of the entire cluster’ since it is not only for storage the metadata of the cluster, but also scheduling and load balancing regions in the TiKV cluster. Besides, The PD server at least three nodes which leads to high availability. It is recommended to deploy PD as an odd number of nodes.
 
-3. Locating the TiKV address for storing and computing data through Placement Driver (PD)
+## Storage servers
```
Comment (lilin90) [inline architecture.md:27]:
```suggestion
## Storage servers

```

**Comment by lilin90** [comment]
@uglyengineer Please fix the image link which causes lint failure.

File: architecture.md (hunk 1 above)
Comment (uglyengineer) [reply architecture.md]: renewed to newest architecture pic

File: architecture.md [hunk 4]
```diff
@@ -6,54 +6,30 @@ aliases: ['/docs/dev/architecture/']
...
 
-TiDB Operator empowers TiDB users to deploy and manage TiDB clusters on mainstream cloud infrastructure (Kubernetes).
+## Storage servers
 
-TiDB Operator:
+### TiKV server
 
-+ Combines the best practices of the container orchestration from the cloud-native community with the know-how of TiDB operation and maintenance.
+The TiKV server is responsible for storing data. From an external view, TiKV is a distributed transactional Key-Value storage engine. Region is the basic unit to store data. Each Region stores the data for a particular Key Range which is a left-closed and right-open interval from StartKey to EndKey. There are multiple Regions in each TiKV node. APIs of TiKV provides native supports to distributed transactions as in Key-Value and supports SI (Snapshot Isolation) level isolation in default, which is the core of how TiDB supports distributed transactions at SQL level. After processing SQL, SQL plan will transfer to actual calling of TiKV APIs. As a result, data will be stored in TiKV. Besides, all data will be autometically maintained as multiple replicas (Three replicas in default), which supports high availability and failover naturally.
```
Comment (lilin90) [inline architecture.md:31]:
```suggestion
The TiKV server is responsible for storing data. TiKV is a distributed transactional key-value storage engine. [Region](/glossary.md#regionpeerraft-group) is the basic unit to store data. Each Region stores the data for a particular Key Range which is a left-closed and right-open interval from StartKey to EndKey. Multiple Regions exist in each TiKV node. TiKV APIs provide native support to distributed transactions at the key-value pair level and supports the Snapshot Isolation level isolation by default. This is the core of how TiDB supports distributed transactions at the SQL level. After processing SQL statements, the TiDB server converts the SQL execution plan to the actual call to the TiKV API. Therefore, data is stored in TiKV. All the data in TiKV is automatically maintained in multiple replicas (three replicas by default), so TiKV has native high availability and supports automatic failover.
```

File: architecture.md [hunk 5]
```diff
@@ -6,54 +6,30 @@ aliases: ['/docs/dev/architecture/']
...
 
-+ Capable of quick deployment, mixed deployment among multiple clusters, automatic operation and maintenance, automatic fail-over, etc.
+### TiFlash Server
 
-+ Makes it user-friendly to use and manage TiDB.
+The TiFlash Server is a special kind of storage server. Differs to normal TiKV, TiFlash is a kind of column-oriented storage server which is mainly designed for OLAP scenarios.
```
Comment (lilin90) [inline architecture.md:35]:
```suggestion
The TiFlash Server is a special type of storage server. Unlike ordinary TiKV nodes, TiFlash stores data by column, mainly designed to accelerate analytical processing.
```

File: architecture.md [hunk 6]
```diff
@@ -6,54 +6,32 @@ aliases: ['/docs/dev/architecture/']
...
+## Storage servers
 
-TiDB Operator:
+### TiKV server
 
-+ Combines the best practices of the container orchestration from the cloud-native community with the know-how of TiDB operation and maintenance.
+The TiKV server is responsible for storing data. TiKV is a distributed transactional key-value storage engine. [Region](/glossary.md#regionpeerraft-group) is the basic unit to store data. Each Region stores the data for a particular Key Range which is a left-closed and right-open interval from StartKey to EndKey. Multiple Regions exist in each TiKV node. TiKV APIs provide native support to distributed transactions at the key-value pair level and supports the Snapshot Isolation level isolation by default. This is the core of how TiDB supports distributed transactions at the SQL level. After processing SQL statements, the TiDB server converts the SQL execution plan to the actual call to the TiKV API. Therefore, data is stored in TiKV. All the data in TiKV is automatically maintained in multiple replicas (three replicas by default), so TiKV has native high availability and supports automatic failover.
```
Comment (lilin90) [inline architecture.md:33]:
```suggestion
The TiKV server is responsible for storing data. TiKV is a distributed transactional key-value storage engine. [Region](/glossary.md#regionpeerraft-group) is the basic unit to store data. Each Region stores the data for a particular Key Range which is a left-closed and right-open interval from StartKey to EndKey. Multiple Regions exist in each TiKV node. TiKV APIs provide native support to distributed transactions at the key-value pair level and supports the Snapshot Isolation level isolation by default. This is the core of how TiDB supports distributed transactions at the SQL level. After processing SQL statements, the TiDB server converts the SQL execution plan to an actual call to the TiKV API. Therefore, data is stored in TiKV. All the data in TiKV is automatically maintained in multiple replicas (three replicas by default), so TiKV has native high availability and supports automatic failover.
```

File: TOC.md [hunk 7]
```diff
@@ -189,7 +189,8 @@
     + [Quick Start](/get-started-with-tispark.md)
     + [User Guide](/tispark-overview.md)
 + Reference
-  + [Architecture](/architecture.md)
+  + Cluster Architecture
+    + [Overview](/architecture.md)
```
Comment (ran-huang) [inline TOC.md:193]: @lilin90 @uglyengineer The corresponding Chinese file is `tidb-architecture.md`. Different file names might bring inconvenience to later maintenance. Is this by design?

Comment (yikeke) [reply TOC.md:193]: Suggest renaming it to tidb-architecture.md to align the Chinese doc. @lilin90 @uglyengineer

Comment (ran-huang) [reply TOC.md:193]: Sure. Updated.

**Pull request merged**